src/sentry/models/apiapplication.py

```diff
@@ -76,7 +75,7 @@ def delete(self, **kwargs):
 
         # There is no foreign key relationship so we have to manually cascade.
         NotificationSetting.objects.remove_for_project(self)
-        with transaction.atomic(), in_test_psql_role_override("postgres"):
+        with outbox_context(transaction.atomic(), kwargs, flush=False):
```

Comment:
What's the purpose of passing `kwargs` through the context?

Reply:
An idea i was toying with but am removing.